### ISSUE TITLE
`mergeProps` 'name' Bug

### ISSUE THREAD

**Issue by korgon** [searchspring/snap#1211]
There is a bug in the `mergeProps` "filterSelectors" function that is causing issues with named components.

Assertion that fails but should not in the `mergeProps` test:

```
describe('filterSelectors function', () => {
	it('removes selectors not within the treePath', () => {
		const treePath = `recommendationBundle BundledCTA price.price-msrp`;

		const selectors = {
			pagination: {},
			icon: {},
			'price': {},
			'search price': {},
			'autocomplete price': {},
			'price.bundle-strike-price': {},
			'recommendationBundle carousel result price': {},
			'recommendationBundle price.bundle-strike-price': {},
			'recommendationBundle price.bundle-price': {},
		};
		const expected = [
			'price',
		];

		expect(filterSelectors(selectors, treePath)).toEqual(expected);
	});
}
```

**Comment by chrisFrazier77** [comment]
https://github.com/searchspring/snap-1.0/pull/24

**Comment by korgon** [comment]
Resolved and merged into `develop` of 1.0 fork.
